docs/my_documentation.rst

```diff
@@ -0,0 +1,544 @@
+
+.. code-block:: python
+
+	MULTIPASS_LOGIN_SELECTOR_TEMPLATE = 'login_selector.html'
+	
+	
+This configuration can be added to your flask configuration file that you use when initializing flask application. (as shown in < Initialization>) However, you can configure multipass also directly through application object. for example:
+
+.. code-block:: python
+
+	app.config['MULTIPASS_LOGIN_SELECTOR_TEMPLATE'] =  'login_selector.html'
+
+Login
+---------------------
+The easiest way to process login is to specify your prefered login URLs in configuration as value for 	``MULTIPASS_LOGIN_URLS``. For example:   
+
+.. code-block:: python
+
+	MULTIPASS_LOGIN_URLS = {'/my_login/', '/my_login/<provider>'}
+	
+(By default,  values ``'/login/'`` and ``'/login/<provider>'`` are set for ``MULTIPASS_LOGIN_URLS`` so if those suits your application there's no need to change them.)
+	
+Multipass then binds  ``process_login`` method to these URLs upon initialization of multipass.
+
+However,  if you wish to execute some additional code before, make ``MULTIPASS_LOGIN_URLS`` empty,  handle login request yourself and then call ``process_login``on your  ``Multipass`` object:
+
+.. code-block:: python
+
+	@app.route('/my_login/', methods=('GET', 'POST'))
+	@app.route('/my_login/<provider>', methods=('GET', 'POST'))
+	def login(provider=None):
+	    # Your additional code
+	    return multipass.process_login(provider)
+	
+
+When calling ``process_login`` with provider name as an argument the ``initiate_external_login`` on the provider gets called and redirects user to the provider's site to perform login.( In case of local provider <``_login_form`` link to login of local providers> gets called.) If provider is not specified the ``process_login`` redirects to a login selection template specified in configuration as ``MULTIPASS_LOGIN_SELECTOR_TEMPLATE``. 
+
+Identification
+-------------------------
+
+After successful authentication response received from ``initiate_external_login``, method ``handle_auth_success`` gets called.
+
+``handle_auth_success`` then collects the identities linked to the user and stores the name of the provider that was used to login  as ``'_multipass_login_provider'`` in ``session``, it will be used when logging out.
+
+Then, the method registered via ``'@multipass.identity_handler'`` decorator is called with one, or list of ``IdentityInfo`` objects as an argument. (Depending whether ``MULTIPASS_ALL_MATCHING_IDENTITIES`` is set in configuration).
+
+Following is our example of ``identity_handler`` method. First, we search through table of identities for one matching ``IdentityInfo.identifier`` from argument. If the identity is not found, we check if we already have this user in the database, we create a new user if needed and assign a new identity to the user object. Then we save this user (or the one found) to the session.
+ 
+ 
+ ???? Should I show user and identity classes from example so its more clear how the code works?
```

Comment:
I don't think it's needed, as long as we still provide the full example code elsewhere.